Rubberduck.VBEEditor/ComManagement/ComSafe.cs

```diff
@@ -0,0 +1,57 @@
+﻿using System;
+using System.Collections.Concurrent;
+using Rubberduck.VBEditor.SafeComWrappers.Abstract;
+
+namespace Rubberduck.VBEditor.ComManagement
+{
+    public class ComSafe: IComSafe
+    {
+        //We override the equality comparison and hash code because subclasses of SafeComWrapper<T> override the corresponding methods.
+        //We need to distinguish between the individual wrapper instances no matter whether they are semantically equal.
+        private readonly ConcurrentDictionary<ISafeComWrapper, byte> _comWrapperCache = new ConcurrentDictionary<ISafeComWrapper, byte>(new ReferenceEqualityComparer());
+
+
+        public void Add(ISafeComWrapper comWrapper)
+        {
+            _comWrapperCache.AddOrUpdate(comWrapper, key => 1, (key, value) => value);
```

Comment:
TBH, this is kind of confusing because we are using `comWrapper` as the key, `key` (which is always `1` as the value. If you do not want to use the hash code for the object as its key, I think a comment is needed here to make clear that we do mean to use the `comWrapper` as the key, not as the value because we don't have a `ConcurrentList<T>` here.

Reply:
This is the standard way to implement a `ConcurrentHashSet`. The actual key in the dictionary is always the result of `equalityComparer.GetHash(key)`.